cmd/clusterctl/hack/local-overrides.py

```diff
@@ -0,0 +1,184 @@
+              'type': 'CoreProvider',
+      },
+      'kubeadm': {
+            'componentsFile': 'bootstrap-components.yaml',
+            'nextVersion': 'v0.3.0',
+            'type': 'BootstrapProvider',
+            'configFolder': '/bootstrap/kubeadm/config/default',
+      },
+}
+
+def load_settings():
+    global settings
+    try:
+        settings = json.load(open('clusterctl-settings.json'))
+    except  Exception as e:
+        raise Exception('failed to load clusterctl-settings.json: {}'.format(e))
+
+def load_provider():
+    provider_repos = settings.get('provider_repos', [])
+    for repo in provider_repos:
+        file = repo + '/clusterctl-settings.json'
+        try:
+            provider_details = json.load(open(file))
+            provider_name = provider_details['name']
+            provider_config = provider_details['config']
+            provider_config['repo'] = repo
+            providers[provider_name] = provider_config
+        except  Exception as e:
+            raise Exception('failed to load clusterctl-settings.json from repo {}: {}'.format(repo, e))
+
+def kustomize(args):
+    try:
+        out = subprocess.Popen(args,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT)
+
+        stdout, stderr = out.communicate()
+        if stderr is not None:
+            raise Exception('stderr contains: \n{}'.format(stderr))
+
+        return stdout
+    except  Exception as e:
+        raise Exception('failed to run {}: {}'.format(args, e))
+
+def get_home():
+    return os.path.expanduser('~')
+
+def write_local_override(provider, version, components_file, components_yaml):
+    try:
+        home = get_home()
+        overrides_folder = os.path.join(home, 'cluster-api', 'overrides')
+        provider_overrides_folder = os.path.join(overrides_folder, provider, version)
+        try:
+            os.makedirs(provider_overrides_folder)
+        except OSError as e:
+            if e.errno != errno.EEXIST:
+                raise
+        f = open(os.path.join(provider_overrides_folder, components_file), 'w')
+        f.write(components_yaml)
+        f.close()
+    except Exception as e:
+        raise Exception('failed to write {} to {}: {}'.format(components_file, components_folder, e))
+
+def create_local_overrides():
+    providerList = settings.get('providers', [])
+    assert providerList is not None, 'invalid configuration: please define the list of providers to overrides'
+    assert len(providerList)>0, 'invalid configuration: please define at least one provider to override'
+
+    for provider in providerList:
+        p = providers.get(provider)
+        assert p is not None, 'invalid configuration: please the configuration for the {} provider'.format(provider)
+
+        repo = '.'
+        if 'repo' in p:
+            repo = p.get('repo')
+
+        config_folder = '/config/default'
+        if 'configFolder' in p:
+            config_folder = p.get('configFolder')
```

Comment:
Can you simplify this to `config_folder = p.get('configFolder, '/config/default')`?

Reply:
done